freqtrade/freqtradebot.py

```diff
@@ -600,8 +600,9 @@ def execute_entry(
         trade_side = 'short' if is_short else 'long'
         pos_adjust = trade is not None
 
+        proposed_leverage = self.config.get('leverage', 1.0)
```

Comment:
rather than adding an additional config property - i would revert the sequence of calls - so "leverage" is called before stake-amount (that was my idea for this problem)

We can take the "assumed" stake-amount (already available based on default settings) - and use that to calculate min/max leverage.
if we look at the leverage brackets - these are usually multiple 1000$ wide - so it's more than doubtful that your default calculation says "20$" - and your custom_stake_amount returns 20.000$ (which would drop it in different max-leverage buckets).

on binance for example, a sample bucket would be 0-5000, 5000-25000, 25_000- 100_000 - so i think the gap is wide enough to take a simple guess.